strawberryfields/gbs/subgraph.py

```diff
@@ -386,3 +389,21 @@ def greedy_degree(
 """dict[str, func]: Included methods for resizing subgraphs. The dictionary keys are strings
 describing the method, while the dictionary values are callable functions corresponding to the
 method."""
+
+
+def is_subgraph(subgraph: Iterable, graph: nx.Graph):
+    """Checks if input is a valid subgraph.
+
+    A valid subgraph is a set of nodes that are contained within the nodes of the graph.
+
+    Args:
+        subgraph (iterable): a collection of nodes
+        graph (nx.Graph): the input graph
+
+    Returns:
+        bool: returns ``True`` only if input subgraph is valid
+    """
+    try:
+        return set(subgraph).issubset(graph.nodes)
+    except TypeError:
+        raise TypeError("subgraph and graph.nodes must be iterable")
```

Comment:
If `graph` is not an `nx.Graph` object, then it will raise an obscure sounding `AttributeError`, and it won't be caught by the try-except. Also, could the `TypeError` be raised by an edge case even when both subgraph and graph.nodes are iterable?

The more I think about it, does this need a try-except at all? Why try and catch this exception, but not any others?

Reply:
Hmmm 🤔 

I'm torn between:
- just directly returning `set(subgraph).issubset(graph.nodes)` in `is_subgraph`
- OR, just delete `is_subgraph` and place the resultant one-liner in the code.

Tempted with the latter, but I'm worried about getting too crazy with deleting functions. E.g., could `is_subgraph` be useful for the user?

Reply:
I would go with the latter as well, unless you have a user story for 'user can determine if a graph contains a specific subgraph'.

I guess it comes down to, is `set(subgraph).issubset(graph.nodes)` simple enough for new users to be able to work out on their own, or not.

Reply:
One of those things where it's easier to make a decision once you have user feedback.

Reply:
Have deleted it, thanks Josh!